app/code/core/Mage/Catalog/Helper/Image.php

```diff
@@ -488,12 +488,12 @@ protected function getWatermarkSize()
     /**
      * Set watermark image opacity
      *
-     * @param int $imageOpacity
+     * @param int|string $imageOpacity
```

Comment:
why adding string if we cast to int later?

Reply:
Because Mage::getStoreConfig would pass string there.

So either let's go back to original concept with cast before Mage::getStoreConfig (let's vote with ❤️) or new one with casting to int inside setWatermarkImageOpacity (let's vote with 🎉).

Reply:
There are several places where Mage::getStoreConfig is cast to int.

Why not add a helper method similiar to getStoreConfigFlag ... ?

```
public static function getStoreConfigAsInt(string $path, $store = null): int
{
    return (int) self::getStoreConfig($path, $store);
}
```

Reply:
@kiatng @fballiano pls share your opinion :)

Reply:
> why adding string if we cast to int later?

I think adding string would be correct, b/c string is valid too.

Without it code analyze may say "unnecessary casting too int" b/c from Docs only int is expected.

Reply:
Sorry for late reply, been very busy. `getStoreConfigAsInt()` and `getStoreConfigAsFloat()` is a good idea. I will support a PR for it. However, I have reservation on imposing strict type on existing functions such as `public function setWatermarkImageOpacity(int $imageOpacity)`, it'll break BC. 

I think adding string to `@param int|string $imageOpacity` is correct.